test/performance/range/views/view_translate_2D_1D_benchmark.cpp

```diff
@@ -0,0 +1,224 @@
+#include <seqan3/range/views/translate_join.hpp>
+#include <seqan3/range/views/to.hpp>
+#include <seqan3/range/views/join.hpp>
+#include <seqan3/test/performance/sequence_generator.hpp>
+#include <seqan3/test/seqan2.hpp>
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+
+#ifdef SEQAN3_HAS_SEQAN2
+    #include <seqan/sequence.h>
+    #include <seqan/seq_io.h>
+    #include <seqan/translation.h>
+#endif
+
+using namespace seqan3;
+using namespace seqan3::test;
+
+struct baseline_tag{};
+struct translate_tag{};
+struct translate_join_tag{};
+
+// ============================================================================
+//  sequential_read
+// ============================================================================
+
+template <typename rng_t>
+void sequential_read_impl(benchmark::State & state, rng_t && rng)
+{
+    for (auto _ : state)
+    {
+        for (auto && outer : rng)
+        {
+            for (aa27 inner : outer)
+            {
+                benchmark::DoNotOptimize(inner.to_rank());
+            }
+        }
+    }
+}
+
+template <typename tag_t>
+void sequential_read(benchmark::State & state)
+{
+    std::vector<std::vector<dna4>> c;
+    c.resize(1000);
+
+    for (size_t i = 0; i < c.size(); ++i)
+        c[i] = test::generate_sequence<seqan3::dna4>(100, 0, 0);
+
+    if constexpr (std::is_same_v<tag_t, baseline_tag>)
+    {
+        std::vector<std::vector<aa27>> cpy = c | views::translate_join | views::to<std::vector<std::vector<aa27>>>;
+        sequential_read_impl(state, cpy);
+    }
+    else if constexpr (std::is_same_v<tag_t, translate_tag>)
+    {
+        auto v = c | views::translate | views::join;
+        sequential_read_impl(state, v);
+    }
+    else
+    {
+        auto v = c | views::translate_join;
+        sequential_read_impl(state, v);
+    }
+}
+
+BENCHMARK_TEMPLATE(sequential_read, baseline_tag);
+BENCHMARK_TEMPLATE(sequential_read, translate_tag);
+BENCHMARK_TEMPLATE(sequential_read, translate_join_tag);
+
+// ============================================================================
+//  random_access
+// ============================================================================
+
+template <typename rng_t>
+void random_access_impl(benchmark::State & state, rng_t && rng,
+                        std::vector<size_t> const & access_positions_outer,
+                        std::vector<size_t> const & access_positions_inner)
+{
+    for (auto _ : state)
+    {
+        for (auto outer : access_positions_outer)
+        {
+            for (auto inner : access_positions_inner)
+            {
+                benchmark::DoNotOptimize(rng[outer][inner].to_rank());
+            }
+        }
+    }
+}
+
+template <typename tag_t>
+void random_access(benchmark::State & state)
+{
+    std::vector<std::vector<dna4>> c;
+    c.resize(1000);
+
+    for (size_t i = 0; i < c.size(); ++i)
+        c[i] = test::generate_sequence<seqan3::dna4>(200, 0, 0);
+
+    std::vector<size_t> access_positions_outer;
```

Comment:
```suggestion
    std::vector<size_t> access_positions_outer{200};
```

Reply:
This does not create a sequence of length 200 but a sequence of length 1 with the element `200`!!!

Never use brace-init with containers unless you are passing the initial values!

Reply:
So should I ignore all the comments concerning brace-init with vector or stringset?

Reply:
As to the point above regarding our style guide, this of course is truly a weird thing and only broken because of the initialiser_list constructor I believe. Indeed `vector x(200)` and `vector x{200}` do completely different things, whereas `std::vector x{200, std::allocator<size_t>{}}` does again the same as the first version using parenthesis 🤕 
I love it. 

Since the latter case with the allocator is even less easy to follow, I agree on using default initialisation followed by an explicit resize.

Reply:
```suggestion
    std::vector<size_t> access_positions_outer{};
```